### DISCUSSION TITLE
Sanctum SPA Authentication: Log Out Other Devices

### DISCUSSION THREAD

**Discussion by jove0610** [laravel/framework#40830] [Ideas]
It doesn't look like this feature isn't available on Sanctum SPA Authentication, the one I saw in docs 
`Auth::logoutOtherDevices($currentPassword);`
needs to uncomment the AuthenticateSession middleware in web:
```
'web' => [
    // ...
    \Illuminate\Session\Middleware\AuthenticateSession::class,
    // ...
],
```
So I googled how to implement the same thing in Sanctum SPA Authentication and found [this youtube video.](url) The instruction is pretty simple: 
1. Create your own middleware (ex. AuthenticateSessionSPA) .
2. Copy everything in the AuthenticateSession and paste it in your new middleware.
3. Create a private variable which is equal to sanctum (ex. private $driver = 'sanctum').
4. Replace all 
`$this->auth->getDefaultDriver()`
with 
`$this->driver`

And just like that I can not use Auth::logoutOtherDevices($password) in my app. The thing is this is not intended feature of laravel spa sanctum so even though it works, I'm not confident if it's safe. I mean, it is pretty simple, right? So maybe there's e a reason why the laravel devs did not implement that? Maybe it has some bugs and security risk?

**Comment by jove0610** [comment]
EDIT: changed the category from **Q&A** to **Ideas**

**Comment by htv-quanth** [comment]
Has the problem been solved?
